src/test/java/org/dz/solution/task32/Task32Test.java

```diff
@@ -0,0 +1,174 @@
+    public void case1() throws IOException {
+        Files.writeString(Path.of(SOURCE_DATA_FILE), "1 3 4 6 9 10 12 13 14 16 19 20");
+        calculateSkippedNumbers();
+        String actualResult = Files.readString(Path.of(RESULT_FILE));
+        assertEquals("2\n5\n7\n8\n11\n15\n17\n18", actualResult);
+    }
+
+    @Test
+    public void case2() throws IOException {
+        Files.writeString(Path.of(SOURCE_DATA_FILE), "1 3 4 12 13 14 16 19 20");
+        calculateSkippedNumbers();
+        String actualResult = Files.readString(Path.of(RESULT_FILE));
+        assertEquals("2\n5\n6\n7\n8\n9\n10\n11\n15\n17\n18", actualResult);
+    }
+
+    @Test
+    public void case3() throws IOException {
+        Files.writeString(Path.of(SOURCE_DATA_FILE), "1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20");
+        calculateSkippedNumbers();
+        String actualResult = Files.readString(Path.of(RESULT_FILE));
+        assertEquals("", actualResult);
+    }
+
+    @Test
+    public void case4() throws IOException {
+        Files.writeString(Path.of(SOURCE_DATA_FILE), "1 3 4 6 13 14");
+        calculateSkippedNumbers();
+        String actualResult = Files.readString(Path.of(RESULT_FILE));
+        assertEquals("2\n5\n7\n8\n9\n10\n11\n12\n15\n16\n17\n18\n19\n20", actualResult);
+    }
+
+    @Test
+    public void case5() throws IOException {
+        Files.writeString(Path.of(SOURCE_DATA_FILE), "1");
+        calculateSkippedNumbers();
+        String actualResult = Files.readString(Path.of(RESULT_FILE));
+        assertEquals("2\n3\n4\n5\n6\n7\n8\n9\n10\n11\n12\n13\n14\n15\n16\n17\n18\n19\n20", actualResult);
+    }
+
+    @Test
+    public void case6() throws IOException {
+        Files.writeString(Path.of(SOURCE_DATA_FILE), "20");
+        calculateSkippedNumbers();
+        String actualResult = Files.readString(Path.of(RESULT_FILE));
+        assertEquals("1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n11\n12\n13\n14\n15\n16\n17\n18\n19", actualResult);
+    }
+
+    private void calculateSkippedNumbers() {
+        int countNumbers = findTheNumberOfNumbersInAFile(SOURCE_DATA_FILE);
+        int[] numbersFromFile = writeNumbersToArray(countNumbers, SOURCE_DATA_FILE);
+        int[] skippedNumbers = findTheMissingNumbers(numbersFromFile);
+        writeArrayToFile(skippedNumbers, RESULT_FILE);
+    }
+
+    private int findTheNumberOfNumbersInAFile(String sourceDataFile) {
+        int count = 0;
+        try (BufferedReader br = new BufferedReader(new FileReader(sourceDataFile))) {
+            String line = br.readLine();
+            if (line == null) {
+                return 0;
+            }
+            String[] words = line.split(WORD_DELIMITER);
+            for (int i = 0; i < words.length; i++) {
+                if (words.length > count) {
+                    count++;
+                }
+            }
+            System.out.println(count);
+        } catch (IOException e) {
+            throw new RuntimeException("Array creation failed", e);
```

Comment:
Senseless error message. There is not an array in this method.
"Counting numbers failed"